### ISSUE TITLE
basho_bench tests directory seems to be ignored  [JIRA: TOOLS-100]

### ISSUE THREAD

**Issue by l1x** [basho/basho_bench#178]
I have installed basho_bench the first time using RPM (previously i used it the source only). I was running into this issue:

```bash
$ basho_bench -d tests /etc/basho_bench/http.config
escript: exception error: no match of right hand side value {error,eacces}
  in function  basho_bench:test_dir/2 (src/basho_bench.erl, line 178)
  in call from basho_bench:main/1 (src/basho_bench.erl, line 47)
  in call from escript:run/2 (escript.erl, line 747)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_it/1
  in call from init:start_em/1
```

I was wondering what is going ton and after poking around I have seen https://github.com/basho/basho_bench/issues/56.

I have tried it with sudo.

```bash
[vagrant@localhost ~]$ sudo basho_bench -d tests /etc/basho_bench/http.config
16:45:47.105 [debug] Lager installed handler {lager_file_backend,
                            "/usr/lib64/basho_bench/lib/basho_bench-0.10.0/ebin/tests/20150408_164547/error.log"} into lager_event
16:45:47.105 [debug] Lager installed handler {lager_file_backend,
                            "/usr/lib64/basho_bench/lib/basho_bench-0.10.0/ebin/tests/20150408_164547/console.log"} into lager_event
```

I guess the following function prefixes the parameter that was passed in with -d with the CWD that is the "/usr/lib64/basho_bench/lib/basho_bench-0.10.0/ebin" in my case if the path is not a full path.
 
https://github.com/basho/basho_bench/blob/0.10.0/src/basho_bench.erl#L172

It would be nice to make it explicit that the path has to be absolute and relative paths will fail if the user has no access to the CWD of the basho_bench binary.

What do you guys think?

**Comment by slfritchie** [comment]
@l1x What do you think about https://github.com/basho/basho_bench/compare/bug/gh178?expand=1 ?

**Comment by l1x** [comment]
@slfritchie i guess I am trying to understand the intentions here. Currently the function behaves like this:

- if you pass in an absolute path it takes it as it is
- if you pass in a relative path it takes it and prefixes it with CWD (that is the ebin/ folder location on disk)

I guess the code should not silently do that or it should be called out. It seems this is a real issue (even though a low priority one) because I am not the first guy who ran into this.

What you are your thoughts?

**Comment by slfritchie** [comment]
The intent of the diff is to use pattern matching to create a more useful error message when something goes wrong: the path of the file with the I/O error is included in any badmatch error reporting.

**Comment by slfritchie** [comment]
I just merged #185 and will close this issue.  If there's a serious problem, or if there's a PR to alter the behavior further, please reopen.

**Comment by Basho-JIRA** [comment]
Ummm, this GH issue was closed on May 19, 2015. Closing in JIRA. 

***[posted via JIRA by Scott Fritchie]***
